test/contracts/core/modules/exchangeRedeemModule.spec.ts

```diff
@@ -0,0 +1,374 @@
+require('module-alias/register');
+
+import * as _ from 'lodash';
+import * as ABIDecoder from 'abi-decoder';
+import * as chai from 'chai';
+import * as setProtocolUtils from 'set-protocol-utils';
+import {
+  Address,
+  Bytes,
+  ExchangeInteractData,
+  KyberTrade,
+  ZeroExSignedFillOrder
+} from 'set-protocol-utils';
+import { BigNumber } from 'bignumber.js';
+
+import ChaiSetup from '@utils/chaiSetup';
+import { BigNumberSetup } from '@utils/bigNumberSetup';
+import {
+  CoreContract,
+  ExchangeRedeemModuleContract,
+  SetTokenContract,
+  SetTokenFactoryContract,
+  StandardTokenMockContract,
+  TransferProxyContract,
+  VaultContract
+} from '@utils/contracts';
+import { ether } from '@utils/units';
+import { assertTokenBalanceAsync, expectRevertError } from '@utils/tokenAssertions';
+import { Blockchain } from '@utils/blockchain';
+import { DEFAULT_GAS } from '@utils/constants';
+import { LogExchangeRedeem } from '@utils/contract_logs/exchangeRedeemModule';
+import { getWeb3 } from '@utils/web3Helper';
+
+import { ExchangeWrapper } from '@utils/wrappers/exchangeWrapper';
+import { CoreWrapper } from '@utils/wrappers/coreWrapper';
+import { ERC20Wrapper } from '@utils/wrappers/erc20Wrapper';
+
+BigNumberSetup.configure();
+ChaiSetup.configure();
+const web3 = getWeb3();
+const Core = artifacts.require('Core');
+const ExchangeRedeemModule = artifacts.require('ExchangeRedeemModule');
+const { SetProtocolTestUtils: SetTestUtils, SetProtocolUtils: SetUtils } = setProtocolUtils;
+const blockchain = new Blockchain(web3);
+const setTestUtils = new SetTestUtils(web3);
+const setUtils = new SetUtils(web3);
+const { expect } = chai;
+const { NULL_ADDRESS, ZERO } = SetUtils.CONSTANTS;
+
+contract('ExchangeRedeemModule', accounts => {
+  const [
+    contractDeployer,
+    zeroExOrderMaker,
+    exchangeRedeemCaller,
+  ] = accounts;
+
+  let core: CoreContract;
+  let transferProxy: TransferProxyContract;
+  let vault: VaultContract;
+  let exchangeRedeemModule: ExchangeRedeemModuleContract;
+  let setTokenFactory: SetTokenFactoryContract;
+
+  const coreWrapper = new CoreWrapper(contractDeployer, contractDeployer);
+  const erc20Wrapper = new ERC20Wrapper(contractDeployer);
+  const exchangeWrapper = new ExchangeWrapper(contractDeployer);
+
+  before(async () => {
+    ABIDecoder.addABI(Core.abi);
+    ABIDecoder.addABI(ExchangeRedeemModule.abi);
```

Comment:
Can we add multiple ABIs? I cant find it but I swear in the past there was a rule that we could only add one.

Reply:
Yeah i believe so. Creation and Checking exchange redeem logs are working